### PR TITLE
[IA-4074] Update version of Werkzueg

### PR DESCRIPTION
**Pull request by n8landolt**
Our current installation of Werkzueg has a vulnerability. We need to update it.

Additionally, switching to Poetry as a package manager will make future updates to vulnerable packages easier.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,7 +19,8 @@ COPY . /work
 WORKDIR /work
 
 # Set up python env
-RUN pip install -r requirements-min.txt && pip install gunicorn
+RUN poetry install
+# RUN pip install -r requirements-min.txt && pip install gunicorn
```
Comment (jdcanas) [inline Dockerfile]: remove commented line?

**Comment by jdcanas** [comment]
Why do we still have a requirements.txt if we are using what seems to be a new poetry config? There are probably some readme commands that need updating...

**Comment by n8landolt** [comment]
Moved back to draft, not quite ready

File: Dockerfile [hunk 2]
```diff
@@ -19,7 +19,9 @@ COPY . /work
 WORKDIR /work
 
 # Set up python env
-RUN pip install -r requirements-min.txt && pip install gunicorn
+RUN pip install "poetry==1.4.0"
+RUN poetry config virtualenvs.create false \
+  && poetry install --no-interaction --no-ansi
```
Comment (lucymcnatt) [inline Dockerfile:24]: what are the diff args here (no-interaction, no-ansi) doing?

File: README.md [hunk 3]
```diff
@@ -105,11 +105,20 @@ gcloud auth login <any-terra-dev-user>
...
+Install dependencies
+```sh
+poetry install
+```
+
+Update dependencies
+```sh
+poetry update
 ```
-This creates a clean virtualenv, installs dependencies from `dependencies-min.txt`, and freezes the resulting environment in `requirements.txt` (which Google App Engine uses during deployment).
```
Comment (lucymcnatt) [inline README.md:112]: might be useful to replace this comment with one explaining what poetry does here?

File: Dockerfile (hunk 2 above)
Comment (n8landolt) [reply Dockerfile:24]: Fairly certain no-interaction is similar to -y, aka it won't look for user input.
no-ansi is supposed to be for easier reading of debug.

File: README.md (hunk 3 above)
Comment (n8landolt) [reply README.md:112]: Sure, I can do that!

Also reading it now... "Google App engine uses during deployment"
Wondering what the implications of my changes are for that...

Comment (lucymcnatt) [reply README.md:112]: hmmmm yeah [this article](https://cloud.google.com/appengine/docs/standard/python3/specifying-dependencies#:~:text=When%20you%20deploy%20to%20App%20Engine%2C%20the%20dependencies%20specified%20in%20the%20requirements.txt%20file%20will%20be%20installed%20automatically%20with%20your%20deployed%20app.) mentions it. I do think there's some poetry command to export to a txt file, that might be necessary...

Comment (lucymcnatt) [reply README.md:112]: https://python-poetry.org/docs/cli/#export:~:text=This%20command%20exports%20the%20lock%20file%20to%20other%20formats.

**Review by jdcanas — APPROVED** [review]

**Comment by jdcanas** [comment]
Maybe update the README deployment here 😬

**Review by lucymcnatt — APPROVED** [review]

**Pull request merged**